### PR TITLE
Fix AppBuilderBase.Start<TMainWindow>

### PR DESCRIPTION
**Pull request by kekekeks**
Also fixes RenderDemo

### CODE REVIEW

**Review by jmacato — APPROVED** [review]

File: src/Avalonia.Controls/AppBuilderBase.cs [hunk 1]
```diff
@@ -126,8 +126,7 @@ public void Start<TMainWindow>(Func<object> dataContextProvider = null)
             
             // Copy-pasted because we can't call extension methods due to generic constraints
             var lifetime = new ClassicDesktopStyleApplicationLifetime(Instance) {ShutdownMode = ShutdownMode.OnMainWindowClose};
```
Comment (donandren) [inline src/Avalonia.Controls/AppBuilderBase.cs]: btw here `Instance` is null and `ClassicDesktopStyleApplicationLifetime` in general is always having a null application.
it's good that  `app.Run(...)` is extension method and everything is working fine, but there might be some general architecture design vulnerabilities

Comment (kekekeks) [reply src/Avalonia.Controls/AppBuilderBase.cs]: That extension method is a compatibility one. The lifetime itself doesn't need the _app at all.

**Comment by grokys** [comment]
@danwalmsley going to merge this. Does this need porting to the 0.9 branch?

**Comment by danwalmsley** [comment]
@grokys go for it
 Ill cherry pick it after.

**Pull request merged**